phpBB/includes/acp/auth.php

```diff
@@ -1139,6 +1139,11 @@ function assign_cat_array(&$category_array, $tpl_cat, $tpl_mask, $ug_id, $forum_
 			@reset($cat_array['permissions']);
 			while (list($permission, $allowed) = each($cat_array['permissions']))
 			{
+				if (!$phpbb_permissions->permission_defined($permission))
```

Comment:
I think you should do the same thing a few lines upper. Because extensions can also add custom categories